### PR TITLE
Fix truncated millisecond timestamp from pino for RFC5424…

### PR DESCRIPTION
**Pull request by dkasten79**
This has been tested on our development device in the lab by modifying the node_modules files directly with the same changes and verified that logs looked correct for both RFC3164 and RFC5424. Syslog server being used is rsyslog on Ubuntu 22.04 LTS.

**RFC3164**
`2025-11-04T17:49:25.915+00:00 jsas-lwc jsas-config[287642]: Getting configuration section network...`

**RFC5424**
`2025-11-04T17:24:41.262Z jsas-lwc jsas-config[286430] Getting configuration section network...`

If you need any other information before merging this change in please let me know.

### CODE REVIEW

File: lib/rfc3164.js [hunk 1]
```diff
@@ -9,7 +9,7 @@ function messageBuilderFactory (options) {
   return function (data) {
     const severity = levelToSeverity.get(data.level)
     const pri = (8 * options.facility) + severity
-    let timestamp = DateTime.fromMillis(data.time, { zone: options.tz }).toFormat('LLL dd HH:mm:ss')
+    let timestamp = DateTime.fromMillis(data.time, { zone: options.tz }).toFormat('LLL dd HH:mm:ss.SSS')
```
Comment (jsumners) [inline lib/rfc3164.js]: This change is not congruent with the spec. Please see https://www.rfc-editor.org/rfc/rfc3164#section-4.1.2.

File: lib/rfc5424.js [hunk 2]
```diff
@@ -10,9 +10,7 @@ function messageBuilderFactory (options) {
   return function (data) {
     const severity = levelToSeverity.get(data.level)
     const pri = (8 * options.facility) + severity
-    const tstamp = DateTime.fromMillis(data.time, { zone: options.tz })
-      .set({ millisecond: 0 })
-      .toISO({ suppressMilliseconds: true })
+    const tstamp = DateTime.fromMillis(data.time, { zone: options.tz }).toISO()
```
Comment (jsumners) [inline lib/rfc5424.js:13]: This will need a test in order to guard against regression.

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "pino-syslog",
-  "version": "3.2.0",
+  "version": "3.2.1-rc.0",
```
Comment (jsumners) [inline package.json]: Please do not update this. We will do so when we release the changes.

**Review by jsumners — CHANGES_REQUESTED** [review]

File: lib/rfc5424.js (hunk 2 above)
Comment (dkasten79) [reply lib/rfc5424.js:13]: All tests will need to be updated as the output will be different for every conversion so regression will be handled through that. I will update all the tests to look for the correct value based on your framework inputs since you do have timestamps in the millisecond range already.

File: package.json (hunk 3 above)
Comment (dkasten79) [reply package.json]: I didn't know how you wanted to handle this so I just changed it to an RC release but I will change this back.

File: lib/rfc3164.js (hunk 1 above)
Comment (dkasten79) [reply lib/rfc3164.js]: I didn't realize that RFC3164 didn't support milliseconds so I will revert this change so thank you for that information. I will revert this change.

File: lib/rfc5424.js (hunk 2 above)
Comment (dkasten79) [reply lib/rfc5424.js:13]: I have updated all tests that I saw using RFC5424 outputs if I missed anything please let me know.

Comment (jsumners) [reply lib/rfc5424.js:13]: Please add a test that specifically asserts that this format requirement is satisfied.

Comment (dkasten79) [reply lib/rfc5424.js:13]: I don't understand why we need a test specific for asserting that the format requirement is satisfied as all the other tests do that. If the format is not satisfied all other tests will fail that use RFC5424 output for certification. And we don't have to worry about there never being a test for RFC5424 as you have tests specific for that output. We already assert the format requirement is satisfied implicitly through all other tests for RFC5424.

Comment (jsumners) [reply lib/rfc5424.js:13]: For the exact reason this PR exists: the end result has not been guaranteed by an explicit test.

Comment (dkasten79) [reply lib/rfc5424.js:13]: The reason this PR exists is because your previous tests for RFC5424 were incorrect and testing for timestamps with no milliseconds in the output which they have all been updated now to check for milliseconds. This is not a configurable feature so there doesn't need to be a specific test for this. I don't even know what you have in mind for a specific test for this because the current tests already test the timestamp format. If you want a specific test for this then go ahead and add one yourself as I would just copy one of your current tests and it would just be a duplicate test at that point and have no real value. I don't know what you are looking for here so please add any tests you feel are needed.

Comment (jsumners) [reply lib/rfc5424.js:13]: We need a test like:

```js
test('timestamps include millisecond precision', () => {
	const result = rfc5424('some log')
	assert.equal(result.timestamp.slice(-4), ".123")
})
```

Comment (dkasten79) [reply lib/rfc5424.js:13]: Ok then add the test you want as I said. I will not be adding the test myself for three reasons:

1. I have never used tap which is what you are using for testing and I am not going to waste time learning it just to add a test I don't agree with being needed.
2. I don't have a setup to run and test the commits to confirm my test is working properly since adding this test is more than just trivial changes in the code.
3. I don't agree with adding this test and it just duplicates something already being tested by all your other tests. All your other tests confirm the timestamp as being there with millisecond precision or else they would all fail.
 
If you want this duplicate test then add it yourself.

Comment (mcollina) [reply lib/rfc5424.js:13]: > Ok then add the test you want as I said.

No. Our primary goal is to keep our libraries maintainable and prevent future regressions.

I recommend you to be a more constructive member of this community.

Comment (dkasten79) [reply lib/rfc5424.js:13]: I am being constructive here and have had nothing but attacks against me since I opened this issue and PR when I asked for reasoning behind why things were done like the stripping of the milliseconds from the timestamp in the first place. I received nothing but non-constructive responses from the repo owner that I was being accusatory when I was just asking a simple question and then given notes that made no sense for the issue mentioning floats when floats are not involved. I was never told why the milliseconds were removed and just dropped it and ended up fixing the issue because I didn't want to waste time arguing over it.

Now it is a test that is a duplicate of every other test that you have for RFC5424. If you give me a valid reason for adding this test like how a regression could happen without the other tests already here failing then give me a scenario where this could happen. I updated every other test in the test framework for RFC5424 so they would pass with the milliseconds added to the timestamp so this alone would test for any regressions because all of those other tests would fail if the timestamp format went back to not having milliseconds on the timestamp.

If you don't want to do this or can't give me a valid scenario where this is needed then the other choice is to add the test you want yourself since it is your project if you really want the test. I have too much going on to add unneeded tests and I won't have a test attached to my name if I don't believe it is needed.

Comment (jsumners) [reply lib/rfc5424.js:13]: Here's my perspective on how we go to this point:

An issue, with a rather demanding and accusatory tone, was opened that fairly ambiguously described a problem. Due to the ambiguity, I assumed the issue was asking about the _input_ data, e.g. a timestamp in the form `123456789.123`, not the output data (`2025-11-09T07:35:00.123`), and provided a response that I think was neutral considering the tone set by the opening post. That received another accusatory and demanding post in response. Given the lack of cooperation, I closed the issue. 

Subsequently, this PR was opened. I gave what I think is a fair review asking for the changes we'd need in order to get it merged and released. Again, I was met with obstinacy instead of cooperation. Yet I continued to provide helpful pointers to get this PR to a mergable state, and continued to receive pushback in response.

-----

Regarding the test, let me try again to be clear:

Our current test suite _does not_ validate that milliseconds are present on the translated date-time string. They passively require it, but, since we don't have one that clearly says "hey, you, the one changing the code, we **want** the fractional seconds component," the next time someone comes along and provides a contribution that touches this code, neither the contributor nor us, the maintainers, have a clear rule that will highlight the breaking change. If the change removes that test, or that test simply does not pass, then we have an easy question to ask: "why was this changed?" In the current state, we just have "oh, well, the tests failed so I massaged them."

-----

My position from here: if the requested test is added, I'll continue supporting this PR to get it merged and released. Otherwise, I will not be bullied in to doing work and will let this PR rot. In that case, if anyone else decides to pick it up, I'll give it due consideration.
